bulk-model-sync-gradle-test/src/test/kotlin/org/modelix/model/sync/gradle/test/PushTest.kt

```diff
@@ -0,0 +1,74 @@
+package org.modelix.model.sync.gradle.test
+
+import GraphLang.L_GraphLang
+import GraphLang.N_Node
+import GraphLang._C_UntypedImpl_Node
+import jetbrains.mps.lang.core.L_jetbrains_mps_lang_core
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Test
+import org.modelix.metamodel.TypedLanguagesRegistry
+import org.modelix.metamodel.typed
+import org.modelix.model.ModelFacade
+import org.modelix.model.api.ConceptReference
+import org.modelix.model.api.IBranch
+import org.modelix.model.api.ILanguageRepository
+import org.modelix.model.api.getDescendants
+import org.modelix.model.api.getRootNode
+import org.modelix.model.client2.ModelClientV2PlatformSpecificBuilder
+import org.modelix.model.client2.getReplicatedModel
+import org.modelix.model.data.ModelData
+import org.modelix.model.data.NodeData
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.mpsadapters.RepositoryLanguage
+import org.modelix.model.server.Main
+import org.modelix.model.sleep
+import org.modelix.model.sync.bulk.asExported
+import java.io.File
+import kotlin.test.assertContentEquals
+
+class PushTest {
+
+    @Test
+    fun `nodes were synced to server`() {
+        val inputDir = File("build/model-sync/testPush")
+        val files = inputDir.listFiles()?.filter { it.extension == "json" } ?: error("no json files found")
+
+        val modules = files.map { ModelData.fromJson(it.readText()) }
+        val inputModel = ModelData(root = NodeData(children = modules.map { it.root }))
+
+        TypedLanguagesRegistry.register(L_GraphLang)
+        TypedLanguagesRegistry.register(L_jetbrains_mps_lang_core)
+        ILanguageRepository.default.registerLanguage(RepositoryLanguage)
+
+        val repoId = RepositoryId("ci-test")
+        val branchName = "master"
+        val url = "http://0.0.0.0:${Main.DEFAULT_PORT}/v2"
+        val branchRef = ModelFacade.createBranchReference(repoId, branchName)
+        val client = ModelClientV2PlatformSpecificBuilder().url(url).build()
+
+        val branch = runBlocking {
+            client.init()
+            client.getReplicatedModel(branchRef).start()
+        }
+        branch.runRead {
+            assertContentEquals(inputModel.root.children, branch.getRootNode().allChildren.map { it.asExported() })
+        }
+
+        applyChangesForPullTest(branch)
```

Comment:
Does this mean the pull test depends on the previous execution of the push test? Can't we break that up. Or if that's really necessary, let's at least express this clearly in the tests by using [test method ordering](https://www.baeldung.com/junit-5-test-order).

Reply:
These tests are not meant to be run together, they perform checks during different stages of the integration test.
They are only run as part of the `ci.sh` test script, which orchestrates the integration test.

Reply:
So is there a dependency between the tests or not? ;-)